src/Polynomials.jl

```diff
@@ -571,13 +564,29 @@ polyfit(xs, ys, 2)
 ```
 
 Original by [ggggggggg](https://github.com/Keno/Polynomials.jl/issues/19)
+More robust version by Marek Peca <mp@eltvor.cz>
+    (1. no exponentiation in system matrix construction, 2. QR least squares)
 """
 function polyfit(x, y, n::Int=length(x)-1, sym::Symbol=:x)
     length(x) == length(y) || throw(DomainError)
     1 <= n <= length(x) - 1 || throw(DomainError)
 
-    A = [ float(x[i])^p for i = 1:length(x), p = 0:n ]
-    Poly(A \ y, sym)
+    #
+    # here unsure, whether similar(float(x[1]),...), or similar(x,...)
+    # however similar may yield unwanted surprise in case of e.g. x::Int
+    #
+    A=similar(float(x[1:1]), length(x), n+1)
+    #
+    # TODO: add support for poly coef bitmap
+    # (i.e. polynomial with some terms fixed to zero)
+    #
+    A[:,1]=1
+    for i=1:n
+        A[:,i+1]=A[:,i] .* x   # cumulative product more precise than x.^n
+    end
+    Aqr=qrfact(A)   # returns QR object, not a matrix
+    p=Aqr\y         # least squares solution via QR
+    Poly(p, sym)
 end
 polyfit(x,y,sym::Symbol) = polyfit(x,y,length(x)-1, sym)
 
```

Comment:
Okay, this looks good to me. If no one chimes in otherwise during the next day or so, I'll merge. Thanks!